src/Session.cpp

```diff
@@ -1180,6 +1180,9 @@ PLAYLIST::CAdaptationSet* SESSION::CSession::DetermineDefaultAdpSet()
       CODEC::FOURCC_HEVC, CODEC::FOURCC_AV01, CODEC::NAME_AV1,    CODEC::FOURCC_VP09,
       CODEC::NAME_VP9,    CODEC::FOURCC_AVC_, CODEC::FOURCC_H264};
 
+  CAdaptationSet* overrideAdp{nullptr}; // Override done by manifest custom parameter
```

Comment:
This can be simplified to return `currAdp` immediately if it is default, without needing to continue the loops:

```
  CAdaptationSet* defaultAdp{nullptr}; // Default determined by codec order

  for (auto& codecCC : videoCodecOrder)
  {
    CAdaptationSet* currAdp{nullptr};
    uint32_t adpIndex{0};
    while ((currAdp = m_adaptiveTree->GetAdaptationSet(adpIndex++)))
    {
      if (currAdp->GetRepresentations().empty() || currAdp->GetStreamType() != StreamType::VIDEO)
        continue;

      if (currAdp->IsDefault())
      {
        return currAdp;
      }
      else if (CODEC::Contains(currAdp->GetCodecs(), codecCC) && !defaultAdp)
      {
        defaultAdp = currAdp;
      }
    }
  }

  return defaultAdp;
```

Reply:
yeah also one less variable, change done
good to go?

Reply:
Sorry, while the other changes seem to make sense don't have a machine/connectivity to properly runtime test. Only really checked the adaptation set default selection on initial playback start.

There have been some reports of both DRM and non-DRM streams failing to play with v22.2.0 but no logs so I don't know what may be causing it.

Reply:
you don't need to test each change, just just to know for your youtube case was ok